### PR TITLE
Implement postgresql-10 support

### PR DESCRIPTION
**Pull request by CyberDem0n**
Mainly it handles rename of xlog to wal.
In the API and inside DCS it is still named xlog (for compatibility).

### CODE REVIEW

File: patroni/postgresql.py [hunk 1]
```diff
@@ -151,23 +151,35 @@ def _configuration_to_save(self):
...
-        return self._use_slots and self._major_version >= 9.4
+        return self._use_slots and self._major_version >= 90400
 
     @property
     def callback(self):
         return self.config.get('callbacks') or {}
 
+    @staticmethod
+    def _wal_name(version):
+        return version >= 100000 and 'wal' or 'xlog'
```
Comment (a1exsh) [inline patroni/postgresql.py]: I think the more widely accepted python way to do this is with: `'xxx' if condition else 'yyy'`

File: patroni/postgresql.py [hunk 2]
```diff
@@ -181,10 +193,10 @@ def get_server_parameters(self, config):
                     parameters.pop('synchronous_standby_names', None)
             else:
                 parameters['synchronous_standby_names'] = self._synchronous_standby_names
-        if self._major_version >= 9.6 and parameters['wal_level'] == 'hot_standby':
+        if self._major_version >= 90600 and parameters['wal_level'] == 'hot_standby':
             parameters['wal_level'] = 'replica'
         return {k: v for k, v in parameters.items() if not self._major_version or
-                self._major_version >= self.CMDLINE_OPTIONS.get(k, (0, 1, 9.1))[2]}
+                self._major_version >= self.CMDLINE_OPTIONS.get(k, (0, 1, 90400))[2]}
```
Comment (a1exsh) [inline patroni/postgresql.py]: Is the change from 9.1 to 9.4 intentional here?

File: patroni/postgresql.py [hunk 3]
```diff
@@ -63,20 +63,20 @@ class Postgresql(object):
...
-        'max_connections': (100, lambda v: int(v) >= 100, 9.1),
-        'max_wal_senders': (5, lambda v: int(v) >= 5, 9.1),
-        'wal_keep_segments': (8, lambda v: int(v) >= 8, 9.1),
-        'max_prepared_transactions': (0, lambda v: int(v) >= 0, 9.1),
-        'max_locks_per_transaction': (64, lambda v: int(v) >= 64, 9.1),
-        'track_commit_timestamp': ('off', lambda v: parse_bool(v) is not None, 9.5),
-        'max_replication_slots': (5, lambda v: int(v) >= 5, 9.4),
-        'max_worker_processes': (8, lambda v: int(v) >= 8, 9.4),
-        'wal_log_hints': ('on', lambda _: False, 9.4)
+        'listen_addresses': (None, lambda _: False, 90100),
```
Comment (a1exsh) [inline patroni/postgresql.py]: Do we really have to change version encoding from float to int here to support v10?

File: patroni/postgresql.py (hunk 1 above)
Comment (Jan-M) [inline patroni/postgresql.py]: Yes, please change. Interesting though that this returns a String

File: patroni/postgresql.py (hunk 3 above)
Comment (CyberDem0n) [inline patroni/postgresql.py]: Yes, inversion is more consistent :)

**Comment by Jan-M** [comment]
:+1:

**Comment by alexeyklyukin** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Comment by a1exsh** [comment]
:+1:

**Pull request merged**